onnxruntime/core/providers/webnn/builders/helper.h

```diff
@@ -302,16 +305,37 @@ static const InlinedHashMap<std::string, std::string> op_map = {
     {"Xor", "logicalXor"},
 };
 
-inline bool CheckSingleOp(const std::string& op_type, const emscripten::val& wnn_builder,
-                          const WebnnDeviceType device_type) {
-  auto op_map_entry = op_map.find(op_type);
-  // Returns false if the op_type is not listed in the op_map or
-  // if the WebNN op has not been implemented in MLGraphBuilder in current browser.
-  if (op_map_entry == op_map.end() || !wnn_builder[op_map_entry->second].as<bool>()) {
-    return false;
-  }
+// WebNN op name to its first input name mapping, only record the name that is different from "input".
+// This map is used to determine the first input name of a WebNN op and is utilized by OpSupportLimits.
+static const InlinedHashMap<std::string, std::string> webnn_op_first_input_name_map = {
```

Comment:
(consider) All static const initializers add to DLL load time (or rather WASM module initialization), even if never used - death by a thousand paper cuts. Consider populating the `std::map` on WebNN EP usage and declaring a simple constexpr array here.

Also can the map be `std::string_view` -> `std::string_view` since they are constants anyway? The `.at()` might not working directly if assigning `const auto& op_type = node.OpType();`, but it should if doing `std::string_view op_type = node.OpType(); ... decomposed_op_map.at(op_type) ...`

Reply:
> (consider) All static const initializers add to DLL load time (or rather WASM module initialization), even if never used - death by a thousand paper cuts. Consider populating the std::map on WebNN EP usage and declaring a simple constexpr array here.

BTW, good learning, thanks!

Reply:
> Do you mean change the vector in this map `InlinedHashMap<std::string, std::vector<std::string>> decomposed_op_map` to constexpr array? But the size of decomposed ops is not fixed.

Oh, using `InlinedHashMap` is fine. I just thought the WebNN EP had a singleton or class instance somewhere that creating the `InlinedHashMap`/`std::map` might make sense, rather than a global static. It was just a consideration though - feel free to reject it (you do know this code better after all).